src/components/preferences/Webhooks.js

```diff
@@ -61,6 +63,12 @@ export default function Webhooks(props) {
         },
     });
 
+    React.useEffect(() => {
+        const hasURL = values?.webhook?.url ? true : false;
+        setEnableTestButton(hasURL && !isTesting);
+        setFieldTouched("webhook.type.type", hasURL, false);
+    }, [isTesting, setFieldTouched, values]);
+
```

Comment:
This seems ok to me, but I'll defer to @psarando since he probably has the most experience with Formik.  

An alternative might be to add a [validate](https://formik.org/docs/api/field#validate) prop to the url field which does this same logic. That way, the logic should only run when the url value changes instead of when any value in `values` changes.

Reply:
I think it's acceptable to only show the validation error message if the user actually touches the Type field, or attempts to save the form without the Type set (and as far as I can tell the Type isn't required for testing the URL).

If we set the `required` prop on the Type field based on the URL field's value, then I don't think we need to manually set this field as "touched" here.